### PR TITLE
Support paged kv cache for single batch chat module

### PR DESCRIPTION
**Pull request by cyx-6**
This PR adds the support for paged kv cache for single batch chat module.

### CODE REVIEW

**Comment by tqchen** [comment]
cc @MasterJH5574 . @cyx-6 would be great to benchmark the speed difference before/after

File: cpp/llm_chat.cc [hunk 1]
```diff
@@ -581,7 +606,16 @@ class LLMChat {
     // Step 5. Load params in nd-array cache.
     this->params_ = ft_.LoadParams(model_path, device_, use_presharded_weights_);
     // Step 6. KV cache creation.
-    this->kv_cache_ = ft_.create_kv_cache_func_();
+    if (ft_.use_paged_kv_cache) {
+      this->kv_cache_config_ = serve::KVCacheConfig(kv_config_json, this->max_window_size_);
```
[the remark this answers is not in the document]
Comment (cyx-6) [reply cpp/llm_chat.cc]: rebased and fiexed

**Comment by tqchen** [comment]
cc @CharlieFRuan on webllm runtime update

**Comment by CharlieFRuan** [comment]
This PR breaks llama on ROCm (single GPU):

Compiling and running `llama-2-7b-q4f16_1` runs into:

```
  File "/home/cfruan/tvm-unity/src/runtime/relax_vm/paged_kv_cache.cc", line 725
TVMError: Check failed: total_seq_length == qkv_data->shape[0] (68 vs. 38) :
```

Compiling and running `llama-2-7b-q4f32_1` runs into `Segmentation fault (core dumped)` with no extra info.

Confirmed that no issues is observed if rebased to https://github.com/mlc-ai/mlc-llm/pull/1627/.

**Pull request merged**